config/filter.d/mysqld-auth.conf

```diff
@@ -17,7 +17,7 @@ before = common.conf
 
 _daemon = mysqld
 
-failregex = ^%(__prefix_line)s(?:\d+ |\d{6} \s?\d{1,2}:\d{2}:\d{2} )?\[Warning\] Access denied for user '\w+'@'<HOST>' (to database '[^']*'|\(using password: (YES|NO)\))*\s*$
+failregex = ^%(__prefix_line)s(?:\d+ |\d{6} \s?\d{1,2}:\d{2}:\d{2} )?\[\w+\] Access denied for user '[^']+'@'<HOST>' 
 
```

Comment:
We indeed have talked before on this, and in that case I agreed since number of possible suffixes was growing large and hard to manage.  It was not the case here.
My overall philosophy could be described here as 'better be safe than sorry'.  Fixing up regexes across all releases of Debian/Ubuntu when we have discovered possibility of DDoS via injection was more costly than staying "safe".  So I would really prefer to be "safe" even if it would cost a tiny bit more upfront by needing to adjust regexes whenever new suffix gets discovered.

And my main fear is that without matching a full line we might fall victim again in case e.g. if underlying service manages to log user-provided input at the beginning of the log string in some log message which we simply didn't observe.  To that service it would be a no bigger -- it is just a log, thus who cares...  For us it would be a major pain! We simply can't assure that this will not happen, we can only assume that there always will be some prefix, and that is what you heavily rely upon by stripping down trailing anchoring.
Situation could be even more gruesome in case of centralized logs (syslog, etc) since then we still might then react to logs from other services and no service identifier gets logged (which I believe we allow to be optional since again out of our control).
So, overall -- I would still prefer to specify failregexes as stringently as possible as  a low-cost preventative measure to avoid us possible major pain and suffering in the future.

Here, the "crime" is even two-fold: you have stripped it during fixing something else, without even a changelog note of that.  Ideally fixes should only concern with fixing issue raised by a bug report and have no side-effects, and not "fix" something which is not broken.  FWIW "Updated ..." without description of what for is imho is not informative